src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs

```diff
@@ -150,7 +155,10 @@ public override IEnumerable<CompletionItem> GetCompletions(CompletionContext con
             }
             GlobalArgs args = new GlobalArgs(this, context.ParseResult);
             IEngineEnvironmentSettings environmentSettings = CreateEnvironmentSettings(args, context.ParseResult);
-            return GetCompletions(context, environmentSettings);
+            var result = GetCompletions(context, environmentSettings);
```

Comment:
After a second thought, I have a concern about disposing here. If calling ParseResult.GetCompletions().ToArray() twice or more, at the second time ObjectDisposedException will happen. I'm not sure if there is such usage externally. If yes, it may crash something. Ideally externally calling it should store the result to a variable instead of calling multiple times.

Reply:
There are few options here:
a) remove the environmentSettings from the `GetCompletions` as it's internal and not used anyways
```
protected internal virtual IEnumerable<CompletionItem> GetCompletions(CompletionContext context)
```
b) Greedy evaluate here in `GetCompletions` and return the materialized sequence (e.g. List) which as well returns IEnumerable, but the underlying enumerations won't be called again on subsequent enumerations. Dispose the environemnt before that
```
using IEngineEnvironmentSettings environmentSettings = CreateEnvironmentSettings(args, context.ParseResult);
return GetCompletions(context, environmentSettings).ToList();
```
c) Lazy enumerate, but make sure we can execute code after accessing last item - either by creating custom wrapper of `IEnumerable` or by yielding:
```
using IEngineEnvironmentSettings environmentSettings = CreateEnvironmentSettings(args, context.ParseResult);
foreach(var item in GetCompletions(context, environmentSettings))
{
    yield return item;
}
```


I'd personaly vote for getting rid of that unused member alltogether (a))

Reply:
Option a) is not feasible since override in derived classes use environment settings. Now I really understood what Vlada means ToList()/ToArray() here(same as option b)).